src/core/services/rest/rest.service.ts

```diff
@@ -189,28 +189,41 @@ export class RestService extends AbstractCore implements EventBusEnabled, Fabric
         const requestHeadersObject = new Headers(requestHeaders);
         const requestInit: RequestInit = this.generateRequestInitObject(restObject, requestHeadersObject);
 
-        const uri: string = this.globalBaseUri + restObject.uri;
-        this.log.debug(`Rest Service: Preparing Fetch Request for URI: ${uri}`, this.getName());
+        let httpRequest;
+
+        // try to create fetch request.
+        try {
+            const uri: string = this.globalBaseUri + restObject.uri;
+            this.log.debug(`Rest Service: Preparing Fetch Request for URI: ${uri}`, this.getName());
+            httpRequest = new Request(uri);
+        } catch (e) {
+            this.log.error(`Rest Service: Cannot create request: ${e}`, this.getName());
```

Comment:
can you add a unit test case for rest.service.ts that captures this branch?

Reply:
Test case added!